### PR TITLE
[Polls] Added an option to display results as progress bars

### PR DESCRIPTION
**Pull request by hawezo**
Polls' results are now optionally displayable as progress bars. The option, `UseProgressBars`, is enabled by default.

Here's a preview:

 

Additionally, some messages have been updated (they are now less exclamative).

### CODE REVIEW

File: module_poll.lua [hunk 1]
```diff
@@ -10,18 +10,39 @@ local enums = discordia.enums
...
+		for _ = 1, choiceProgressLength do
+			progressText = progressText .. progressCharacter
+		end
+
+		-- Fill void
+		for _ = 1, progressLength - choiceProgressLength do
+			progressText = progressText .. progressVoidCharacter
+		end
+
+		return string.format('`[' .. progressText .. ']` **%d**   %s (%d%%)', choiceVotes, voteText, percentage)
```
[the remark this answers is not in the document]
Comment (hawezo) [reply module_poll.lua]: My bad, inattention issue

File: module_poll.lua [hunk 2]
```diff
@@ -30,24 +51,21 @@ function Module:GetPollFooter(member, duration, isResults)
...
-	end
+
+	local verb = isResults and "Lasts" or "Lasted"
 
 	if duration < 60 then
 		duration = 60
 	end
 
-	text = text .. string.format(". %s for %s.", verb, util.FormatTime(duration))        
+	text = string.format("%s. %s for %s.", text, verb, util.FormatTime(duration))
```
[the remark this answers is not in the document]
Comment (hawezo) [reply module_poll.lua:61]: I understand but I think it's clearer this way. When I read it first I didn't realized it was added next to `text`, but this version makes it clear.

File: module_poll.lua [hunk 3]
```diff
@@ -454,58 +481,59 @@ function Module:IsAChoice(poll, emoji)
...
-	
+	local title = preview and "[Preview] " .. poll.title or poll.title
+
 	for i, choice in ipairs(poll.choices) do
 		if Bot:GetEmojiData(guild, choice.emoji.Name) ~= nil then
-			table.insert(fields, { 
+			table.insert(fields, {
 				name = "Choice n°" .. i,
-				value = string.format("%s %s", choice.emoji.MentionString, choice.text)
+				value = string.format("%s  %s", choice.emoji.MentionString, choice.text)
```
[the remark this answers is not in the document]
Comment (hawezo) [reply module_poll.lua:501]: **Without**: 
 

**With**:

File: module_poll.lua [hunk 4]
```diff
@@ -10,18 +10,39 @@ local enums = discordia.enums
...
+		local progressText = ''
+		local progressLength = 20 -- Determines the length (in characters) of the progress bar
+		local progressCharacter = '='
+		local progressVoidCharacter = ' '
+		local percentage = (choiceVotes / totalVotes) * 100
+		local choiceProgressLength = math.floor((choiceVotes / totalVotes) * progressLength);
+
+		-- Advance progress
+		for _ = 1, choiceProgressLength do
+			progressText = progressText .. progressCharacter
```
Comment (SirLynix) [inline module_poll.lua:33]: https://www.lua.org/manual/5.3/manual.html#pdf-string.rep

**Pull request merged**